src/x25519.rs

```diff
@@ -237,27 +232,27 @@ impl StaticSecret {
     /// Extract this key's bytes for serialization.
     #[inline]
     pub fn to_bytes(&self) -> [u8; 32] {
-        self.0.to_bytes()
+        self.0
```

Comment:
Weird edge case: Previously, this type did clamping on deserialization. So if you loaded some unclamped value `x`, then `to_bytes()` would previously output `clamp(x)`. Similarly, serializing `x` would produce `clamp(x)`

With the proposed changes, this method would just output `x`. This is technically breaking, but seems pretty minor.
 
Solution: Keep the below weird deserialization code to make sure that `StaticSecret` never holds anything but clamped integers.

cc @tarcieri

Reply:
IMO it’s more correct to output the unclamped scalar.

NaCl *only* stores the unclamped scalar for X25519 (and the seed for Ed25519) and does all the clamping on-the-fly.

Reply:
I agree. Ok we can do that and I'll just put a note in the changelog